bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -0,0 +1,244 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.senechome.internal;
+
+import static org.eclipse.smarthome.core.types.RefreshType.REFRESH;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.senechome.internal.json.SenecHomeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SenecHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Steven Schwarznau - Initial contribution
+ */
+@NonNullByDefault
+public class SenecHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SenecHomeHandler.class);
+    private final static String VALUE_TYPE_INT = "u3";
+    private final static String VALUE_TYPE_UNSIGNED_INT = "u8";
+    private final static String VALUE_TYPE_FLOAT = "fl";
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable PowerLimitationStatusDTO limitationStatus = null;
+    private final @Nullable SenecHomeApi senecHomeApi;
+    private SenecHomeConfigurationDTO config = new SenecHomeConfigurationDTO();
+
+    public SenecHomeHandler(Thing thing, SenecHomeApi senecHomeApi) {
+        super(thing);
+        this.senecHomeApi = senecHomeApi;
+    }
+
+    @Override
+    public void handleRemoval() {
+        stopJobIfRunning();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == REFRESH) {
+            logger.debug("Refreshing {}", channelUID);
+            refresh();
```

Comment:
Refresh is called on all channels. So that's a bit of an overkill here. What you can do is to use an `ExpiringCache` with a timeout of like 5 seconds. Something like:

```
private final ExpiringCache<Boolean> refreshCache = new ExpiringCache<>(Duration.ofSeconds(5), this::refreshState);
```

```
public void refresh() {
   refreshCache.getValue();
}

public @Nullable Boolean refreshState() {
   .. code of refresh method
   return Boolean.TRUE;
}
```
This way the state is only updated if the timeout is exceeded. And if multiple calls are done within a short time frame it won't trigger multiple state updates.